appcenter-push/android/src/main/java/com/microsoft/appcenter/reactnative/push/AppCenterReactNativePushModule.java

```diff
@@ -10,18 +10,39 @@
 import com.microsoft.appcenter.push.Push;
 import com.microsoft.appcenter.reactnative.shared.AppCenterReactNativeShared;
 import com.microsoft.appcenter.utils.async.AppCenterConsumer;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
 
 @SuppressWarnings("WeakerAccess")
 public class AppCenterReactNativePushModule extends BaseJavaModule {
 
+    private static final String ENABLE_PUSH_IN_JAVASCRIPT = "enable_push_in_javascript";
+
+    private static final String PUSH_ONCE_ENABLED = "PushOnceEnabled";
+
+    private boolean mPushStarted;
+
     private AppCenterReactNativePushEventListener mPushListener;
 
     public AppCenterReactNativePushModule(Application application) {
         this.mPushListener = new AppCenterReactNativePushEventListener();
         Push.setListener(mPushListener);
         AppCenterReactNativeShared.configureAppCenter(application);
         if (AppCenter.isConfigured()) {
-            AppCenter.start(Push.class);
+            boolean startPush = true;
+            boolean enablePushInJavascript = AppCenterReactNativeShared.getConfiguration().optBoolean(ENABLE_PUSH_IN_JAVASCRIPT);
+            if (enablePushInJavascript) {
+
+                /*
+                 * TODO expose a way to post command in background looper in native SDK to avoid accessing storage directly here.
+                 * Storage might not be ready yet.
```

Comment:
What happens if storage is not ready?  Would we crash, or would the push configuration just not be set?

Reply:
We make the storage ready in the next line.